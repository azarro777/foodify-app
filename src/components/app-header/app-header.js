import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import Modal from '../modal/modal';
import './app-header.scss';

export const AppHeader = ({ visible }) => {
  const [modalActive, setModalActive] = useState(false);

  // const location = useLocation();
  // const { pathname } = location;
  // const splitLocation = pathname.split('/');

  return (
    <header className='header'>
      {/* <li className={splitLocation[1] === '' ? 'active' : ''}> */}
      <Link className='header__link' to='/'>
        Random dish
      </Link>
      {/* </li> */}
      {/* <li className={splitLocation[1] === 'favourites' ? 'active' : ''}> */}
      <Link className='header__link' to='/favourites'>
        Favourites
      </Link>
      {/* </li> */}
      <div className={visible}>
        <button className='header__button' onClick={() => setModalActive(true)}>
          Add custom dish
        </button>
      </div>
      <Modal active={modalActive} setActive={setModalActive} />
    </header>
  );
};
