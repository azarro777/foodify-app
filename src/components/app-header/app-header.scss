@mixin text-style {
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 18px;
  line-height: 21px;
  color: #000000;
}

.header {
  background-color: #f2f2f2;
  width: 1440px;
  height: 60px;
  display: flex;
  justify-content: flex-end;
  justify-content: center;
  align-items: center;
  padding: 0 50px;
}

.header li {
  list-style: none;
}

.header__link {
  @include text-style;
  text-decoration: none;
  padding: 0 1.8rem;
  white-space: nowrap;
}

.header__link:hover {
  cursor: pointer;
  color: rgb(104, 212, 176);
}

.active a {
  text-decoration: underline;
}

.isVisible {
  display: none;
}

.button__container {
  position: relative;
}

.header__button {
  position: absolute;
  left: 230px;
  top: -20px;
  width: 192px;
  height: 38px;
  background: #c4c4c4;
  border: none;
  @include text-style;
}

.header__button:hover {
  cursor: pointer;
  background: #68d4b0;
}

@media (max-width: 1024px) {
  .header__button {
    left: 50px;
  }
}

@media (max-width: 414px) {
  .header__button {
    left: 0px;
    width: 50px;
    overflow: hidden;
    @include text-style;
  }

  .header__button::before {
    content: '+';
    display: block;
    width: 100%;
    height: 50%;
    left: 100%;
  }
}

@media (max-width: 320px) {
  .header__button {
    left: 0px;
  }

  .header__link {
    padding: 0 0.5rem;
    white-space: nowrap;
  }
}
