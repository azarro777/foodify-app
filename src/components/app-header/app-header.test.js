import React from 'react';
import { AppHeader } from './app-header';
import { shallow } from 'enzyme';

describe('Testing <AppHeader/>', () => {
  it('AppHeader have rendered correctly', () => {
    const appHeader = shallow(<AppHeader />);
    expect(appHeader).toMatchSnapshot();
  });
});
