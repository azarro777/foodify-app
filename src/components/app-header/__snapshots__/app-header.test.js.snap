// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AppHeader/> AppHeader have rendered correctly 1`] = `ShallowWrapper {}`;
